### PR TITLE
NAS-105604 / 12.0 / Handle edge case where retrieved mac_prefix can be an interface name

### PR DESCRIPTION
**Pull request by sonicaj**
Make sure to follow and check these boxes before submitting a PR! Thank you.

- [ ] Explain the feature
- [ ] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: iocage_lib/ioc_json.py [hunk 1]
```diff
@@ -579,6 +580,12 @@ def get_mac_prefix():
             # Use the hosts prefix to start generation from.
             # Helps avoid clashes with other systems in the network
             mac_prefix = default_mac[0]['addr'].replace(':', '')[:6]
+            if len(mac_prefix) != 6 or not set(mac_prefix).issubset(string.hexdigits):
+                # We do this because in certain cases ( very likely due to netifaces not properly
+                # retrieving values ) mac_prefix can be `lo0` which results in an error
+                # as we consider this to be a valid mac prefix below and just try to replace bits
+                # which in this case don't exist resulting in an unintended exception
+                raise KeyError()
```
Comment (skarekrow) [inline iocage_lib/ioc_json.py]: Can't we raise a nicer error?

**Review by skarekrow — CHANGES_REQUESTED** [review]

File: iocage_lib/ioc_json.py (hunk 1 above)
Comment (sonicaj) [reply iocage_lib/ioc_json.py]: Sure we can, thoughts on `ValueError` ?

Comment (skarekrow) [reply iocage_lib/ioc_json.py]: I think `ValueError` is more appropriate here, but I meant a message explaining what the comment does. They'll hit this and not look in the code for *why*.

Comment (sonicaj) [reply iocage_lib/ioc_json.py]: So as in the title, this is an edge case and a fault in `netifaces` which we have patched locally. I don't think we should raise an error here, if we are unable to get the mac address of default gateway, that's okay, we can just generate our own. Thoughts ?

Comment (skarekrow) [reply iocage_lib/ioc_json.py]: Fair enough :)

Comment (sonicaj) [reply iocage_lib/ioc_json.py]: Perfect ;)

**Review by skarekrow — APPROVED** [review]

**Comment by skarekrow** [comment]
Thanks!

**Comment by sonicaj** [comment]
No problem ;)

**Pull request merged**